config/clients/js/template/client.mustache

```diff
@@ -0,0 +1,482 @@
+   * @param {string} [paginationOptions.continuationToken]
+   * @param {*} [options] Override http request option.
+   * @throws { FgaError }
+   */
+  async listStores(options: ClientRequestOptsWithAuthZModelId & PaginationOptions = {}): PromiseResult<ListStoresResponse> {
+    return this.api.listStores(options.pageSize, options.continuationToken, options);
+  }
+
+  /**
+   * CreateStore - Initializes a store
+   * @param body
+   * @param options
+   */
+  async createStore(body: CreateStoreRequest, options: ClientRequestOpts = {}): PromiseResult<CreateStoreResponse> {
+    return this.api.createStore(body, options);
+  }
+
+  /**
+   * GetStore - Returns the current store
+   * @param options
+   */
+  async getStore(options: ClientRequestOpts = {}): PromiseResult<GetStoreResponse> {
+    return this.api.getStore(options);
+  }
+
+  /**
+   * DeleteStore - Deletes a store
+   * @param options
+   */
+  async deleteStore(options: ClientRequestOpts = {}): PromiseResult<void> {
+    return this.api.deleteStore(options);
+  }
+
+  /************************
+   * Authorization Models *
+   ************************/
+
+  /**
+   * ReadAuthorizationModels - Reads all authorization models
+   * @param options
+   */
+  async readAuthorizationModels(options: ClientRequestOpts = {}): PromiseResult<ReadAuthorizationModelsResponse> {
+    return this.api.readAuthorizationModels();
+  }
+
+  /**
+   * WriteAuthorizationModel - Creates a new version of the authorization model
+   * @param body
+   * @param options
+   */
+  async writeAuthorizationModel(body: WriteAuthorizationModelRequest, options: ClientRequestOpts = {}): PromiseResult<WriteAuthorizationModelResponse> {
+    return this.api.writeAuthorizationModel(body, options);
+  }
+
+  /**
+   * ReadAuthorizationModel - Read the current authorization model
+   * @param options
+   */
+  async readAuthorizationModel(options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<ReadAuthorizationModelResponse> {
+    const authorizationModelId = this.getAuthorizationModelId(options);
+    if (!authorizationModelId) {
+      throw new Error("authorization_model_id_required");
+    }
+    return this.api.readAuthorizationModel(authorizationModelId, options);
+  }
+
+  /**
+   * ReadLatestAuthorizationModel - Reads the latest authorization model for the current store
+   * @param options
+   */
+  async readLatestAuthorizationModel(options: ClientRequestOpts = {}): PromiseResult<ReadAuthorizationModelResponse> {
+    const { headers = {} } = options;
+    setHeaderIfNotSet(headers, CLIENT_METHOD_HEADER, "ReadLatestAuthorizationModel");
+    const authorizationModelsResponse = await this.readAuthorizationModels({ ...options, headers });
+    const response = authorizationModelsResponse as any as CallResult<ReadAuthorizationModelResponse>;
+    response.authorization_model = authorizationModelsResponse.authorization_models?.[0];
+    delete (response as any).authorization_models;
```

Comment:
This is calling `ReadAuthorizationModels`, which has all the models. This method is only concerned with the latest model.

Reply:
Right, what I meant is why do we need to delete the authroziation_models as oppose to just return the  authorizationModelsResponse.authorization_models?.[0];

Reply:
There's other elements in the response we want to return (e.g. raw response). So instead of copying them over to a new object, we just remove that from the existing object.